### ISSUE TITLE
add_hexagon_layer() doesn't seem to work correctly with weighting variable

### ISSUE THREAD

**Issue by atsokol** [anthonynorth/rdeck#25]
Hi Anthony,

I was trying to use your rdeck for doing some plotting work. It offers extra functionality compared to mapdeck, which would be very convenient for me (tooltips, layer customization, etc.). However, there seems to be an error in how add_hexagon_layer() function operates with weighting variables. Could you please look into it?

I provide a reproducible example below. Data is taken from COVID19 package.

```{r}
library(tidyverse)
library(COVID19)
library(paletteer)
library(rdeck)
library(sf)

df_2 <- covid19("Germany", level = 3, start=ymd(20201231), end=ymd(20201231)) %>%
  filter(!is.na(latitude), !is.na(latitude)) %>%
  mutate(position = sfc_point(longitude, latitude))

# Hexagon layer with weighting variable doesn't work 
(hexagon_1 <- 
    rdeck(map_style = mapbox_light(),
      theme='light',
      initial_bounds = st_bbox(df_2$position),
      initial_view_state = view_state(pitch=45)
      ) %>%
    add_hexagon_layer(data = df_2,
                    get_color_weight = deaths,
                    get_elevation_weight = deaths,
                    position_format = "XY",
                    radius = 10000,
                    color_range = paletteer_d("RColorBrewer::YlOrBr", 6),
                    pickable = TRUE,
                    extruded = TRUE,
                    tooltip = c(administrative_area_level_3))
)

# Hexagon layer without weighting variable works just fine
(hexagon_2 <- 
    rdeck(map_style = mapbox_light(),
      theme='light',
      initial_bounds = st_bbox(df_2$position),
      initial_view_state = view_state(pitch=45)
      ) %>%
    add_hexagon_layer(data = df_2,
                    # get_color_weight = deaths,
                    # get_elevation_weight = deaths,
                    position_format = "XY",
                    radius = 10000,
                    color_range = paletteer_d("RColorBrewer::YlOrBr", 6),
                    pickable = TRUE,
                    extruded = TRUE,
                    tooltip = c(administrative_area_level_3))
)

# For comparison, heatmap layer with weighting variable works just fine
(heatmap <- 
rdeck(map_style = mapbox_light(),
      theme='light',
      initial_bounds = st_bbox(df_2$position)) %>%
  add_heatmap_layer(data = df_2,
                        get_weight = deaths,
                        opacity = 0.3,
                        pickable = TRUE,
                        color_range = paletteer_d("RColorBrewer::YlOrBr", 6))
)
```

**Comment by anthonynorth** [comment]
This appears to be a bug with deck.gl HexagonLayer; specifically it doesn't support these accessors with the serialisation format rdeck uses.

As a workaround until this is fixed in deck.gl, you could use a h3 hexagon layer to create a 3d hexmap.

```r
# remotes::install_github("obrl-soil/h3jsr")
library(h3jsr)

covid_hexes <- df_2 %>%
  mutate(hexagon = point_to_h3(position, 5)) %>%
  group_by(hexagon) %>%
  summarise(
    deaths = sum(deaths, na.rm = TRUE),
    # concat for the tooltip
    administrative_area_level_3 = paste0(administrative_area_level_3, collapse = "\n"),
    .groups = "drop"
  )

rdeck(
  map_style = mapbox_light(),
  theme = "light",
  initial_bounds = st_bbox(df_2$position),
  initial_view_state = view_state(pitch = 45)
) %>%
  add_h3_hexagon_layer(
    data = covid_hexes,
    get_elevation = scale_linear(col = deaths),
    elevation_scale = 25000,
    get_fill_color = scale_color_linear(
      col = deaths,
      palette = paletteer_d("RColorBrewer::YlOrBr", 6)
    ),
    pickable = TRUE,
    extruded = TRUE,
    tooltip = c(administrative_area_level_3)
  )
```

![image](https://user-images.githubusercontent.com/391385/109463660-2300ff00-7ab1-11eb-8ccf-9f46cb9cdfdb.png)

**Comment by atsokol** [comment]
Thank you. It worked.

**Comment by anthonynorth** [comment]
This was patched in deck.gl 8.4.0 and will be fixed in rdeck 0.4.0.
